### PR TITLE
make haproxy rate limiting configurable

### LINKED ISSUE #2418: haproxy ACLs: allow to disable the abuse ACLs
Some people prefer user experience over improved performance and can prefer disabling the abuser ACL in haproxy especially since it returns a simple 403 without many explanations which can cause calls to their help desk.

We should provide an option to disable these ACLs in haproxy

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
make haproxy rate limiting configurable and add a more meaningful message when a user is rate-limited

# Impacts
haproxy rate limiting

# Issue
fixes #2418 

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Allowed haproxy captive portal rate-limiting to be configurable

### CODE REVIEW

**Comment by fdurand** [comment]
It's optional but it should be better to remove that from the generated haproxy config if the rate limiting is disabled.
 
`       acl action var(req.action) -m found
        acl unflag_abuser src_clr_gpc0 --
        http-request allow if action unflag_abuser
        http-request deny if { src_get_gpc0 gt 0 }
`

**Comment by julsemaan** [comment]
@dwlfrth also made me notice:
https://github.com/inverse-inc/packetfence/blob/devel/lib/pf/services/manager/haproxy.pm#L159

**Comment by julsemaan** [comment]
@fdurand + @dwlfrth 

I think I've fixed all the required places, please confirm

**Comment by fdurand** [comment]
Looks good to me

File: lib/pf/services/manager/haproxy.pm [hunk 1]
```diff
@@ -30,6 +30,7 @@ use pf::file_paths qw(
     $install_dir
     $conf_dir
     $var_dir
+    $captiveportal_templates_path
```
Comment (dwlfrth) [inline lib/pf/services/manager/haproxy.pm:33]: I think you are missing the `pf::file_paths` path !

**Review by dwlfrth — DISMISSED** [review]

File: lib/pf/services/manager/haproxy.pm (hunk 1 above)
Comment (dwlfrth) [reply lib/pf/services/manager/haproxy.pm:33]: Forget it !

**Review by dwlfrth — APPROVED** [review]
LGTM

**Pull request merged**